entity_parser.go

```diff
@@ -333,35 +335,72 @@ func parseNameTag(tag, defaultName string) (string, string, error) {
 	return fullNameTag, name, nil
 }
 
+// parseETLTag functions parses DOSA "etl" tag
+func parseETLTag(tag string) (string, ETLState, error) {
+	fullETLTag := ""
+	etlTag := ""
+	matches := etlPattern0.FindStringSubmatch(tag)
+	if len(matches) == 2 {
+		fullETLTag = matches[0]
+		etlTag = matches[1]
+	}
+
+	if len(matches) == 0 {
+		return "", EtlOff, nil
+	}
+
+	// filter out "trailing comma"
+	etlTag = strings.TrimRight(etlTag, " ,")
+
+	var err error
+	etlTag, err = NormalizeName(etlTag)
+	if err != nil {
+		return "", EtlOff, err
+	}
+
+	etlState, err := ToETLState(etlTag)
+	if err != nil {
+		return "", EtlOff, err
+	}
+	return fullETLTag, etlState, nil
+}
+
 // parseEntityTag function parses DOSA tag on the "Entity" field
-func parseEntityTag(structName, dosaAnnotation string) (string, *PrimaryKey, error) {
+func parseEntityTag(structName, dosaAnnotation string) (string, ETLState, *PrimaryKey, error) {
 	tag := dosaAnnotation
+
+	// find the ETL flag
```

Comment:
since the ETL flag is optional , so let's parse the required tag such as primarykey first.

Reply:
Basically, I'd like to append the `etl` tag at the end, as it's more natural for users to add this to their new schema. However, due to the way we parse `primaryKey`, if we parse it first, it'll treat the entire substring (including `etl`) as primary key. Therefore I parse the tag backwards, i.e. `etl`->`primaryKey`->`name`.

We could change the parsing order by changing the way we parse primaryKey though, which is a bit out of the scope of this pull.

Reply:
The reason to parse required tag first is to give user the most critical error first.  If the etl is part of the primarykey some how, we want to indicate the error of it.  If there is a field is called by the optional tag name such as 'name', 'etl' . we want to parse those first before we parse optional tag.

Reply:
Being that said, we should include tests to have field name `etl` and `name`

Reply:
done